packages/core/src/__tests__/unit/application.unit.ts

```diff
@@ -305,15 +285,63 @@ describe('Application', () => {
       expect(binding.key).to.equal('services.my-service');
       expect(findKeysByTag(app, 'service')).to.containEql(binding.key);
     });
+  });
 
-    function givenApp() {
-      app = new Application();
+  describe('shutdown signal listener', () => {
+    it('registers a SIGTERM listener when app starts', async () => {
+      const count = getListeners().length;
+      await app.start();
+      expect(getListeners().length).to.eql(count + 1);
+    });
+
+    it('does not impact SIGTERM listener when app stops without start', async () => {
+      const count = getListeners().length;
+      await app.stop();
+      expect(getListeners().length).to.eql(count);
+    });
+
+    it('registers/removes a SIGTERM listener by start/stop', async () => {
+      await app.start();
+      const count = getListeners().length;
+      await app.stop();
+      expect(getListeners().length).to.eql(count - 1);
+      // Restart
+      await app.start();
+      expect(getListeners().length).to.eql(count);
+    });
+
```

Comment:
Please add a test to verify that `new Application()` does not register any new listeners.